flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/RuntimeMonitorHandler.java

```diff
@@ -113,7 +117,17 @@ private void respondAsLeader(ChannelHandlerContext ctx, Routed routed, ActorGate
 		DefaultFullHttpResponse response;
 
 		try {
-			String result = handler.handleRequest(routed.pathParams(), jobManager);
+			Map<String, String> params = routed.pathParams();
```

Comment:
If you pass two maps (routed + query) then this is much less effort per request, and keeps the types of parameters nicely separated.

Also: Why do you need the query parameter here? This handler is the base for REST style calls, it should never need query parameters.